### PR TITLE
refactor(material-experimental/mdc-table): share code between MDC and non-MDC versions of MatTableDataSource

### PR DESCRIPTION
**Pull request by mmalerba**
blocked on https://github.com/angular/components/pull/20924

### CODE REVIEW

File: src/material-experimental/mdc-table/text-column.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * By default, the name of this column will be the header text and data property accessor.
+ * The header text can be overridden with the `headerText` input. Cell values can be overridden with
+ * the `dataAccessor` input. Change the text justification to the start or end using the `justify`
+ * input.
+ */
+@Component({
+  selector: 'mat-text-column',
+  template: `
+    <ng-container matColumnDef>
```
Comment (crisbeto) [inline src/material-experimental/mdc-table/text-column.ts:24]: For the other MDC components we also share the template string by exporting it through `material/table`. Can we do it for this component too?

File: src/material/table/table-data-source.ts [hunk 2]
```diff
@@ -27,20 +28,16 @@ import {map} from 'rxjs/operators';
...
- * which defines how row data is converted to a string for filter matching.
- *
- * **Note:** This class is meant to be a simple data source to help you get started. As such
- * it isn't equipped to handle some more advanced cases like robust i18n support or server-side
- * interactions. If your app needs to support more advanced use cases, consider implementing your
- * own `DataSource`.
- */
-export class MatTableDataSource<T> extends DataSource<T> {
+interface Paginator {
+  page: EventEmitter<PageEvent>;
```
Comment (crisbeto) [inline src/material/table/table-data-source.ts:32]: Maybe we should make this a `Subject` so it's Angular-agnostic? `EventEmitter` extends `Subject` already.

Comment (mmalerba) [reply src/material/table/table-data-source.ts:32]: Sure, I don't think its super important though, this is just an internal thing. The API we expose to users is still `MatPaginator`

**Comment by mmalerba** [comment]
feedback addressed

**Review by crisbeto — APPROVED** [review]
Sorry for the delay, I must have missed the notification. LGTM, but there's a lint check failure.

**Review by jelbourn — APPROVED** [review]
LGTM

**Pull request merged**